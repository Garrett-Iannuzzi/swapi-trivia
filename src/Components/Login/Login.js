import React, { Component } from 'react';
import './Login.scss';

class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      quote: '',
      skillLevel: 'Youngling',
      hasError: false
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleLoginError = () => {
    const { name, quote, skillLevel } = this.state;
    if (name === '' || quote === '' || skillLevel === '') {
      this.setState({ hasError: true });
    } else {
      this.setState({ hasError: false });
      this.props.handleLogin();
      this.props.handleUserInfo(this.state);
    }
  }

  submitLoginHandler = () => {
    this.handleLoginError();
  }

  render() {
      return (
        <form className='form-login'>
          <h2>Please Login</h2>
            <input
              type='text'
              name='name'
              value={this.state.name}
              placeholder='Enter Name'
              className='input-login'
              onChange={ (e) => this.handleChange(e) }
            />
            <input
              type='text'
              name='quote'
              value={this.state.quote}
              placeholder='Enter Favorite StarWars Quote'
              className='input-login'
              onChange={ (e) => this.handleChange(e) }
            />
          <div className='input-container'>
            <label for="skill">Select:</label>
            <select
              id='skill'
              value={this.state.skillLevel}
              name='skillLevel'
              className='input-select'
              onChange={ (e) => this.handleChange(e) }
            >
              <option value='Youngling'>Youngling</option>
              <option value='Padawan'>Padawan</option>
              <option value='Jedi Master'>Jedi Master</option>
            </select>
          </div>
          {this.state.hasError && <p className='error-message'>Please Fill out all forms</p>}
          <button className='btn-login' type='button' onClick={ () => this.submitLoginHandler() }>Login</button>
        </form>
      )
  }
}

export default Login;
