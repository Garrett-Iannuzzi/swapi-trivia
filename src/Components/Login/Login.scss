main {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: space-around;
}

img {
  height: 200px;
  width: 350px;
  margin-top: 15px;
  opacity: .85;
}

form {
  height: 400px;
  width: 45%;
  opacity: .8;
  background-color: black;
  display: flex;
  margin: auto;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  border: 3px solid white;
  border-radius: 10px;
}

.nav-logo {
  margin: 15px 0;
  height: 100px;
  width: 175px;
}

.input-login {
  width: 40%;
  font-size: 1em;
  padding: 10px;
  border-radius: 10px;
  border: 0;
  outline: none;
  cursor: pointer;
}

.btn-login {
  border: 3px solid white;
  width: 75%;
  font-size: 1em;
  border-radius: 10px;
  padding: 10px;
  background-color: black;
  color: white;
  outline: 0;
  cursor: pointer;
}

.btn-login:hover {
  background-color: white;
  color: black;
  font-weight: bold;
}

.input-select {
  width: 100%;
  font-size: 1em;
  height: 35px;
  border-radius: 10px;
  border: 0;
  outline: none;
  cursor: pointer;
}

.input-container {
  display: flex;
  flex-direction: column;
  width: 40%
}

.link-button {
  width: 65%;
  display: flex;
  justify-content: center;
}

label {
  color: white;
  margin-bottom: 5px;
}
