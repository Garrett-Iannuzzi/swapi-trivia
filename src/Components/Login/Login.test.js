import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login/Login.js';

describe('Login', () => {

  let wrapper, mockHandleLoginError, mockhandleUserInfo

  beforeEach(() => {
    mockHandleLoginError = jest.fn();
    mockhandleUserInfo = jest.fn();
    wrapper = shallow(
      <Login 
        handleLoginError={mockHandleLoginError}
        handleUserInfo={mockhandleUserInfo}
      />
    )
  });

  it('Should match the snapshot with data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });







})