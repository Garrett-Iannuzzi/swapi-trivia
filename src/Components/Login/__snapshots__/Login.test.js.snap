// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match the snapshot with data passed in 1`] = `
<main
  className="main-login"
>
  <img
    alt="star wars logo"
    className="Login-logo"
    src="starwarslogo.png"
  />
  <form
    className="form-login"
  >
    <h2
      className="header-login"
    >
      Please Login
    </h2>
    <input
      className="input-login"
      maxLength="25"
      name="name"
      onChange={[Function]}
      placeholder="Enter Name"
      type="text"
      value=""
    />
    <input
      className="input-login"
      maxLength="65"
      name="quote"
      onChange={[Function]}
      placeholder="Enter Favorite StarWars Quote"
      type="text"
      value=""
    />
    <div
      className="input-container"
    >
      <label
        for="skill"
      >
        Skill Level:
      </label>
      <select
        className="input-select"
        id="skill"
        name="skillLevel"
        onChange={[Function]}
        value="Youngling"
      >
        <option
          value="Youngling"
        >
          Youngling
        </option>
        <option
          value="Padawan"
        >
          Padawan
        </option>
        <option
          value="Jedi Master"
        >
          Jedi Master
        </option>
      </select>
    </div>
    <Link
      className="link-button"
      to="/movies"
    >
      <button
        className="btn-login"
        onClick={[Function]}
        type="button"
      >
        May The Force Be With You
      </button>
    </Link>
  </form>
</main>
`;
